pkg/flow/internal/controller/node_custom_component.go

```diff
@@ -54,6 +58,29 @@ type CustomComponentNode struct {
 
 var _ ComponentNode = (*CustomComponentNode)(nil)
 
+// ExtractImportAndDeclare extracts an importNamespace and a customComponentName from a componentName.
+// TODO: this function assumes that CustomComponentNames and ImportNamespaces don't contain "."
+// In other words, there are two possible scenarios:
+// - [customComponentName] LABEL ->  instance of a local declare
+// - [importNamespace].[customComponentName] LABEL -> instance of an imported declare
+// To address this limitation in the future, we will need to find a different approach to
+// identify the importNamespaces and CustomComponentNames.
+func ExtractImportAndDeclare(componentName string) (string, string) {
+	parts := strings.Split(componentName, ".")
```

Comment:
At least for now, this will be a little more prepared for the possibility of future changes to custom components if this is changed to use `strings.SplitN(componentName, ".", 2)`, that way only the first `.` is considered for the import namespace, and other `.` can be a part of the declare name.